numpy/ma/core.py

```diff
@@ -371,24 +371,24 @@ def maximum_fill_value(obj):
         raise TypeError(errmsg)
 
 
-def _recursive_set_default_fill_value(dtypedescr):
+def _recursive_set_default_fill_value(dtype):
     deflist = []
-    for currentdescr in dtypedescr:
-        currenttype = currentdescr[1]
-        if isinstance(currenttype, list):
+    for name in dtype.names:
+        currenttype = dtype[name].base
```

Comment:
It is because of subarrays, and I should probably add a comment explaining that. The idea is to drop the subarray part and get just the type, because that is what the old code did:

The old code did `currentdescr = descr[1]`, and remember that `descr` has the form `(name, type, shape)`, so that would drop the shape. In the same way doing `.base` drops the shape.